samples/textDetection.js

```diff
@@ -78,7 +77,7 @@ Index.prototype.add = function(filename, document, callback) {
     self.tokenClient.set(filename, document, cb);
   });
 
-  async.parallel(tasks, callback);
+  Promise.all(tasks).then(callback);
```

Comment:
I think you can just ```return Promise.all(tasks);``` here 
and on line 60 make it into an async function
`Index.prototype.add = async (filename, document) => {`

Reply:
Done ! 
https://github.com/vijay-qlogic/nodejs-vision/pull/1/commits/bb6ccacc6481592a693e8d7c357219f88de87c96